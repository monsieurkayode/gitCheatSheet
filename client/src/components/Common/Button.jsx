import styled from 'styled-components';

export default styled.input`
  text-align: center;
  padding: 10px 40px;
  margin-left: 42px;
  min-width: 200px;
  background: #668090;
  color: #FFF;
  border-radius: 50px;
  font-size: 13px;
  font-weight: 500;
  font-family: Prompt, sans-serif;
  text-transform: uppercase;
  cursor: pointer;
  letter-spacing: 2px;
  
  &:hover {
    background: #546975;
  }
`;
