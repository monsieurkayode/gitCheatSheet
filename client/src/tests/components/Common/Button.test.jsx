import React from 'react';
import { mount } from 'enzyme';
import Button from '../../../components/Common/Button';

describe('Button Component', () => {
  const wrapper = mount(<Button />);

  it('should render correctly', () => {
    expect(wrapper).toMatchSnapshot();
  });
});
