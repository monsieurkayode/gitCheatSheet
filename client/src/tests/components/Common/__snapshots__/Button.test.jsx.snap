// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button Component should render correctly 1`] = `
<styled.input>
  <StyledComponent
    forwardedComponent={
      Object {
        "$$typeof": Symbol(react.forward_ref),
        "attrs": Array [],
        "componentStyle": ComponentStyle {
          "componentId": "sc-bdVaJa",
          "isStatic": true,
          "lastClassName": "kOaxiG",
          "rules": Array [
            "
  text-align: center;
  padding: 10px 40px;
  margin-left: 42px;
  min-width: 200px;
  background: #668090;
  color: #FFF;
  border-radius: 50px;
  font-size: 13px;
  font-weight: 500;
  font-family: Prompt, sans-serif;
  text-transform: uppercase;
  cursor: pointer;
  letter-spacing: 2px;
  
  &:hover:not(:disabled) {
    background: #546975;
  }
",
          ],
        },
        "displayName": "styled.input",
        "foldedComponentIds": Array [],
        "render": [Function],
        "styledComponentId": "sc-bdVaJa",
        "target": "input",
        "toString": [Function],
        "warnTooManyClasses": [Function],
        "withComponent": [Function],
      }
    }
    forwardedRef={null}
  >
    <input
      className="sc-bdVaJa kOaxiG"
    />
  </StyledComponent>
</styled.input>
`;
